pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/configuration/SettingsEditor.kt

```diff
@@ -66,11 +66,23 @@ class SettingsEditor : SettingsEditor<RunConfiguration>() {
     }
 
     override fun resetEditorFrom(runConfiguration: RunConfiguration) {
+        if (!runConfiguration.gradleExecutable.isNullOrEmpty()) {
+            runConfiguration.gradleExecutable.also { gradleExecutableField.text = it ?: "gradlew" }
+        } else {
+            gradleExecutableField.text = "gradlew"
+        }
+        if (!runConfiguration.taskName.isNullOrEmpty()) {
+            gradleTaskField.text = runConfiguration.taskName!!
+        } else {
+            gradleTaskField.setTextAndAddToHistory("pitest")
+        }
         targetClasses.text = runConfiguration.classFQN
-        gradleTaskField.text = runConfiguration.taskName
-        runConfiguration.gradleExecutable.also { gradleExecutableField.text = it ?: "" }
         buildTypesField.text = runConfiguration.buildType
         verboseCheckbox.isSelected = runConfiguration.verbose
+
+        runConfiguration.project.service<PluginCheckerService>().getBuildTypes().forEach { buildType ->
+            buildTypesField.history.add(buildType)
```

Comment:
Did you test this here with the history. Because for me it didn't work. Btw how did you trigger a reset?